apps/emqx/src/emqx_listeners.erl

```diff
@@ -61,8 +64,12 @@
 -export([certs_dir/2]).
 -endif.
 
+-type listener_id() :: atom() | binary().
+
 -define(CONF_KEY_PATH, [listeners, '?', '?']).
 -define(TYPES_STRING, ["tcp", "ssl", "ws", "wss", "quic"]).
+-define(MARK_DEL, marked_for_deletion).
+-define(MARK_DEL_BIN, <<"marked_for_deletion">>).
```

Comment:
Should we emphasize that `?MARK_DEL` is a _command_ (that should never appear in a translated config), while `?MARK_DEL_BIN` is a raw config _value_ preventing population? That is, to have `mark_for_deletion` atom for command and `<<"marked_for_deletion">>` as a raw conf value?

Reply:
Make sense.
I'll update the macro in a follow up PR.

Reply:
Maybe:
`?CONFIG_COMMAND_MARK_DEL`
`?CONFIG_VALUE_MARK_DEL` ?

Reply:
😓 The confusion is that it is unclear from the naming, whether we consider `MARK_DEL` to be a special update command or a special value.
* If we treat it as a real value, then seems that the convention that `update_config` works with either commands or raw configs is violated
* If we treat it as a command, then it is a bit confusing to have command with the same name as value 😅